dashboard/app/models/script_level.rb

```diff
@@ -546,22 +546,33 @@ def self.summarize_as_bonus_for_teacher_panel(script, bonus_level_ids, student)
     end
   end
 
-  # Bring together all the information needed to show the teacher panel on a level
-  def summarize_for_teacher_panel(student)
-    contained_levels = levels.map(&:contained_levels).flatten
-    contained = contained_levels.any?
+  def contained_levels
+    levels.map(&:contained_levels).flatten
+  end
+
+  # Returns the level whose status will determine state of the progress bubble
+  def get_level_for_progress(student)
+    if bubble_choice?
+      keep_working_level = level.keep_working_sublevel(student, script)
```

Comment:
A brief comment here explaining the desired logic might be helpful.

This function is great, it's an important piece of our progress bubble logic.